torchvision/transforms/v2/functional/_utils.py

```diff
@@ -41,16 +45,22 @@ def register_kernel(dispatcher, datapoint_cls):
     return _register_kernel_internal(dispatcher, datapoint_cls, datapoint_wrapper=False)
 
 
-def _get_kernel(dispatcher, datapoint_cls):
+def _get_kernel(dispatcher, input_type):
+    if not issubclass(input_type, (torch.Tensor, PIL.Image.Image)):
+        raise TypeError(
+            f"Dispatcher '{dispatcher}' supports plain tensors, any TorchVision datapoint, or a PIL images, "
+            f"but got {input_type} instead."
+        )
```

Comment:
This is the type error that we previously had on every dispatcher. This makes sure we error in case the user inputs something unsupported in general like a string. Note that this does **not** implement the "error on unsupported types" that we are talking about in https://github.com/pytorch/vision/pull/7747#issuecomment-1661698450.